### DISCUSSION TITLE
How do I set Appearance to my device?

### DISCUSSION THREAD

**Discussion by vishesh-varma** [h2zero/NimBLE-Arduino#493] [Q&A]
I'm not sure where, but I've seen BLE devices detecting as a dedicated keyboard, or mice. I'm assuming that it can be changed by Appearance Characteristic in General Access Profile. How do I go around doing that in esp32's nimBLE implementation? I have tried doing this:
```c++
NimBLEDevice::init(deviceName);
    NimBLEServer *pServer = NimBLEDevice::createServer();
    NimBLEService *infoService = pServer->createService("180A");
    NimBLECharacteristic *firmwareRevision = infoService->createCharacteristic(
        "2A26",
        NIMBLE_PROPERTY::READ);
    NimBLECharacteristic *manuracturerName = infoService->createCharacteristic(
        "2A29",
        NIMBLE_PROPERTY::READ);
    firmwareRevision->setCallbacks(new firmwareRevisionCallback());
    manuracturerName->setCallbacks(new manufacturerNameCallback());
    infoService->start();
    
    NimBLEAdvertising *pAdvertising = NimBLEDevice::getAdvertising();

    pAdvertising->addServiceUUID(infoService->getUUID());
    pAdvertising->setAppearance(0x0083);                                         /////// THIS
    pAdvertising->setScanResponse(true);
    pAdvertising->start();
```
According to [Appearance Values](https://specificationrefs.bluetooth.com/assigned-values/Appearance%20Values.pdf), we know that 0x0083 indicates a Laptop. But, setting appearance with `setAppearance(0x0083)` doesn't do anything to my esp32. It still gets detected an an UNKNOWN DEVICE.

**Comment by h2zero** [comment]
That should work, not sure why it wouldn't. What are you using to scan the device?

**Comment by vishesh-varma** [comment]
This seems to be working with some scanning mobile applications. I was asssuming that I would see `Laptop` in Generic Access section of the device along with the device name. But, reading appearance gives zero there. That being said, the device is being read as a `Laptop` somehow, because I can see the different icon for it in the scanning list in the scanners that have that.
